packages/asset/contracts/RoyaltyHandler.sol

```diff
@@ -0,0 +1,50 @@
+//SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.18;
+
+// initializable
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+
+abstract contract RoyaltyHandler is Initializable {
+    address private royaltyRecipient;
+    mapping(uint256 => uint256) private royaltyBpsByTokenId;
+
+    function __RoyaltyHandler_init(
+        address recipient
+    ) internal onlyInitializing {
+        __RoyaltyHandler_init_unchained(recipient);
+    }
+
+    function __RoyaltyHandler_init_unchained(
+        address recipient
+    ) internal onlyInitializing {
+        royaltyRecipient = recipient;
+    }
+
+    function getRoyaltyBps(uint256 tokenId) internal view returns (uint256) {
+        return royaltyBpsByTokenId[tokenId];
+    }
+
+    function setRoyaltyBps(uint256 tokenId, uint256 royaltyBps) internal {
+        royaltyBpsByTokenId[tokenId] = royaltyBps;
+    }
```

Comment:
i would prefix the function because it's internal, like on OZ
https://github.com/OpenZeppelin/openzeppelin-contracts-upgradeable/blob/master/contracts/token/ERC1155/extensions/ERC1155URIStorageUpgradeable.sol
```
setRoyaltyBps => _setRoyaltyBps
```

Reply:
N/A as per above